web-server/v0.4/mock/api.js

```diff
@@ -1,64 +1,64 @@
 import config from './datastoreConfig';
 
-export const mockControllerAggregation = {
-  aggregations: {
-    controllers: {
-      buckets: [
-        {
-          key: 'a_test_controller',
-          doc_count: 1,
-          runs_preV1: { value: null },
-          runs: { value: 1, value_as_string: '1111-11-11' },
-        },
-        {
-          key: 'b_test_controller',
-          doc_count: 2,
-          runs_preV1: { value: null },
-          runs: { value: 2, value_as_string: '2222-22-22' },
-        },
-      ],
+/* We generate 26 records to generate specific records starting
+ with each alphabet to assert our tests. Hence we have 26 instances
+ of indices and results as per the controllers. */
+
+// Generates mock controllers
+export const generateMockControllerAggregation = () => {
+  const mockControllerSchema = {
+    aggregations: {
+      controllers: {
+        buckets: [],
+      },
     },
-  },
+  };
+  for (let i = 0; i < 26; i += 1) {
+    mockControllerSchema.aggregations.controllers.buckets.push({
+      key: `${(i + 10).toString(36)}_test_controller`,
+      doc_count: i,
+      runs_preV1: { value: null },
+      runs: {
+        value: `${i + 1}`,
+        value_as_string: `${i + 1}${i + 1}${i + 1}${i + 1}-${i + 1}${i + 1}-${i + 1}${i + 1}`,
+      },
+    });
+  }
+  return mockControllerSchema;
 };
 
 const datastoreConfig = config['/dev/datastoreConfig'];
 const prefix = datastoreConfig.prefix + datastoreConfig.run_index.slice(0, -1);
-export const mockIndices = [
-  {
-    index: `${prefix}.0000-00-00`,
-  },
-  {
-    index: `${prefix}.0000-00-01`,
-  },
-];
+// Generates mock indices
+export const generateMockIndices = () => {
+  const indexSchema = [];
+  for (let i = 0; i < 26; i += 1) {
+    indexSchema.push({ index: i < 10 ? `${prefix}.0000-00-0${i}` : `${prefix}.0000-00-${i}` });
+  }
+  return indexSchema;
+};
 
-export const mockResults = {
-  hits: {
-    hits: [
-      {
-        fields: {
-          'run.name': ['a_test_run'],
-          '@metadata.controller_dir': ['test_run.test_domain.com'],
-          'run.start': ['1111-11-11T11:11:11+00:00'],
-          'run.id': ['1111'],
-          'run.end': ['1111-11-11T11:11:12+00:00'],
-          'run.controller': ['test_run.test_domain.com'],
-          'run.config': ['test_size_1'],
-        },
-      },
-      {
-        fields: {
-          'run.name': ['b_test_run'],
-          '@metadata.controller_dir': ['b_test_run.test_domain.com'],
-          'run.start': ['1111-11-11T11:11:13+00:00'],
-          'run.id': ['2222'],
-          'run.end': ['1111-11-11T11:11:14+00:00'],
-          'run.controller': ['b_test_run.test_domain.com'],
-          'run.config': ['test_size_2'],
-        },
+// Generates mock results
+export const generateMockResults = () => {
+  const mockResultsSchema = {
+    hits: {
+      hits: [],
+    },
+  };
+  for (let i = 0; i < 26; i += 1) {
```

Comment:
Further, since we use the number 26 in hard-coded in two places, having it be driven by a constant or derived from the pagination scheme would make this code much more robust.

Reply:
@portante Thanks for your update on this Pull Request. I have updated the number of items to `[10, 20, 50, 100]` now.

I agree that we should not hardcode values as such and generate controllers from the mock. I believe we hardcoded values because the mock objects, ``controllers, indices, results` follow a specific schema as such and there are boundaries on which we can generate these values, for example, we have prefixes to indices and so on. It's a bit tricky to set this up. What I'd like to suggest is for us to have a json api consisting of an array of objects which we can include over here. It would be really helpful if you can share your approach here if I'm missing something. Thanks!

Reply:
@portante I believe you want it to function like an API, for example: If we select say 100 items, this value `100` is passed on to the loop and it generates 100 items then and there. So, basically the `mock.js` has functions that act as API endpoints